src/components/AddressDetailExplorer.js

```diff
@@ -0,0 +1,528 @@
+        });
+        return;
+      }
+
+      // update the query params state then fetch the new page
+      this.setState({ queryParams, loadingHistory: true }, () => {
+        // Fetch new data, unless query params were cleared and we were already in the most recent page
+        this.getHistoryData(+queryParams.page);
+      });
+    }
+  }
+
+  /**
+   * Called when 'network' ws message arrives
+   * If it's a new tx message update data, in case is necessary
+   *
+   * @param {Object} wsData Data from websocket
+   */
+  handleWebsocket = (wsData) => {
+    if (wsData.type === 'network:new_tx_accepted') {
+      if (this.shouldUpdate(wsData, false) && !this.state.warningRefreshPage) {
+        // If the search address is in one of the inputs or outputs
+        this.setState({ warningRefreshPage: true });
+      }
+    }
+  }
+
+  /**
+   * Check if address is valid and then update the state and get data from full node
+   * If not valid show error message
+   *
+   * @param {Object} address New searched address to update state
+   */
+  updateAddress = (address) => {
+    this.setState({ queryParams: this.pagination.obtainQueryParams() }, () => {
+      if (hathorLib.transaction.isAddressValid(address)) {
+        this.setState({
+          address,
+          loadingTokens: true,
+          loadingSummary: true,
+          loadingHistory: true,
+          addressTokens: {},
+          transactions: [],
+          balance: {},
+          errorMessage: '',
+        }, () => {
+          const queryParams = this.pagination.obtainQueryParams();
+          if (queryParams.token !== null) {
+            // User already have a token selected on the URL
+            this.setState({ selectedToken: queryParams.token }, () => {
+              this.reloadData();
+            });
+          } else {
+            this.reloadData();
+          }
+        });
+      } else {
+        this.setState({ errorMessage: 'Invalid address.' });
+      }
+    });
+  }
+
+  /**
+   * Update transactions data state after requesting data from the server
+   *
+   * @param {Object} queryParams URL parameters
+   */
+  getHistoryData = (page) => {
+    // update query parameters
+    const query = this.pagination.obtainQueryParams();
+    if (query.page !== page) {
+      query.page = page;
+      const newURL = this.pagination.setURLParameters(query);
+      this.props.history.push(newURL);
+    }
+
+    return addressApi.getHistory(this.state.address, this.state.selectedToken, TX_COUNT, TX_COUNT * page).then((response) => {
+      if (!response) {
+        // An error happened with the API call
+        this.setState({ showReloadTokenButton: true });
+        return;
+      }
+      const txhistory = response || [];
+      this.setState({ transactions: txhistory }, () => {
+
+        const promises = [];
+        for (const tx of txhistory) {
+          if (!this.state.txCache[tx.tx_id]) {
+            promises.push(txApi.getTransaction(tx.tx_id));
```

Comment:
I would just explain here in details why you need this get. That this is to keep the same column data and in the wallet service you don't have this information.

Explain which column it is and which information you don't have in the wallet service.